external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -95,6 +105,40 @@ impl Packages {
     }
 }
 
+impl CompilerToolchain {
```

Comment:
In general I have some open questions about input around read/write operations on lock file. One question here is if this read/write is reasonable in the `schema.rs` file? vs the `mod.rs` file?

More general question: I'm a bit unsure about how lockfile creation / edits are structured right now. Basically, we have a [temp `LockFile::new` creation](https://github.com/MystenLabs/sui/blob/main/external-crates/move/crates/move-package/src/resolution/dependency_graph.rs#L1040) in the dependency resolution code. If I'm following I think this will need to be pulled out (so that the temp lockfile handle is available for writing outside dep resolution) or some other kind of structuring to edit the lock file [after the dependency changes are committed](https://github.com/MystenLabs/sui/blob/main/external-crates/move/crates/move-package/src/lib.rs#L192-L195).

Reply:
> More general question: I'm a bit unsure about how lockfile creation / edits are structured right now. 

So I think I've dug myself out of this one--my plan is to add a method to copy over the existing lock contents and return a handle to write (append) the compiler versioning separately--it's too much trouble to try preserve and pass around the existing `LockFile::new` handle created for the dependency graph.

Reply:
I'm a little unclear on the details around reading just the `CompilerToolchain` vs reading the whole lock file and then pulling out the `CompilerToolchain` from that, but if it works in testing, then all good.

Re: your other question, your intuition is correct that you either need to find a way to expose that temporary lock file, or write to the lock file afterwards.  No strong preference on which way you go, just be careful not to overwrite the toolchain information in cases where the build failed.